library/packages/src/modules/SignatureCheckDialogs.rb

```diff
@@ -1197,6 +870,350 @@ def UseFileWithUnknownDigest(filename, digest, dont_show_dialog_ident)
     publish function: :ImportGPGKeyIntoTrustedDialog, type: "boolean (map <string, any>, integer)"
     publish function: :UseFileWithWrongDigest, type: "boolean (string, string, string, string)"
     publish function: :UseFileWithUnknownDigest, type: "boolean (string, string, string)"
+
+  private
+
+    # helper to strip download path. It uses internal knowledge that download
+    # prefix ends in TmpDir.* zypp location
+    def strip_download_prefix(path)
+      path.sub(/^\/.*\/TmpDir\.[^\/]+\//, "")
+    end
+
+    def HandleDoNotShowDialogAgain(default_return, dont_show_dialog_ident, dont_show_dialog_checkboxid, dont_show_url)
+      dont_show_status = Convert.to_boolean(
+        UI.QueryWidget(Id(dont_show_dialog_checkboxid), :Value)
+      )
+      # Widget doesn't exist
+      if dont_show_status.nil?
+        Builtins.y2warning(
+          "No such UI widget with ID: %1",
+          dont_show_dialog_checkboxid
+        )
+        # Checkbox selected -> Don't show again
+      elsif dont_show_status == true
+        Builtins.y2debug(
+          "User decision -- don't show the dialog %1 again, setting default return %2",
+          dont_show_dialog_ident,
+          default_return
+        )
+        SetShowThisPopup(dont_show_dialog_ident, false, dont_show_url)
+        SetDefaultDialogReturn(
+          dont_show_dialog_ident,
+          default_return,
+          dont_show_url
+        )
+        # Checkbox not selected -> Show again
+      else
+        SetShowThisPopup(dont_show_dialog_ident, true, dont_show_url)
+      end
+
+      nil
+    end
+
+    # Function adds delimiter between after_chars characters in the string
+    #
+    # @param string to be splitted
+    # @param [String] delimiter
+    # @param integer after characters
+    # @return [String] with delimiters
+    def StringSplitter(whattosplit, delimiter, after_chars)
+      splittedstring = ""
+      after_chars_counter = 0
+      max_size = Builtins.size(whattosplit)
+
+      loop do
+        if Ops.greater_or_equal(
+          Ops.add(after_chars_counter, after_chars),
+          max_size
+        )
+          splittedstring = Ops.add(
+            Ops.add(splittedstring, splittedstring == "" ? "" : delimiter),
+            Builtins.substring(whattosplit, after_chars_counter)
+          )
+          break
+        else
+          splittedstring = Ops.add(
+            Ops.add(splittedstring, splittedstring == "" ? "" : delimiter),
+            Builtins.substring(whattosplit, after_chars_counter, after_chars)
+          )
+          after_chars_counter = Ops.add(after_chars_counter, after_chars)
+        end
+      end
+
+      splittedstring
+    end
+
+    # Returns term with message icon
+    #
+    # @param string message type "error", "warning" or "question"
+    # @return [Yast::Term] `Image(...) with margins
+    def MessageIcon(msg_type)
```

Comment:
Ken requested removing all icons from popups, maybe we should drop this when touching this anyway...